modulemd/modulemd-defaults-v1.c

```diff
@@ -1322,45 +1316,43 @@ modulemd_defaults_v1_merge (const gchar *module_name,
       merged_default_stream =
         g_hash_table_lookup (merged->intent_default_streams, intent_name);
 
-      if (!merged_default_stream)
+      /* If there is no new default stream, just jump to the next item */
+      if (!intent_default_stream)
+        continue;
+
+      else if (!merged_default_stream && intent_default_stream)
```

Comment:
```suggestion
      else if (!merged_default_stream)
```
The `if` above already made sure `intent_default_stream` isn't NULL.